pkg/util/http/calls.go

```diff
@@ -3,17 +3,18 @@ package httputil
 import (
 	"context"
 	"errors"
+	"net"
 	"net/http"
 )
 
-// Serve listens and serves the internal HTTP server.
+// ListenAndServe listens and serves the internal HTTP server.
 //
 // Returns any error returned by the internal server
 // except http.ErrServerClosed.
 //
 // After Shutdown call, Serve has no effect and
 // the returned error is always nil.
-func (x *Server) Serve() error {
+func (x *Server) ListenAndServe() error {
```

Comment:
i think it is possible to implement via the new `Listen` and `Serve` together. but up to you